packages/paste-core/components/badge/src/index.tsx

```diff
@@ -1,49 +1,70 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import {Box} from '@twilio-paste/box';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {secureExternalLink} from '@twilio-paste/anchor';
+import {ErrorIcon} from '@twilio-paste/icons/esm/ErrorIcon';
 
 import type {BadgeProps, BadgeVariants} from './types';
 import {useResizeChildIcons} from './hooks';
-import {badgeFocusableStyles, badgeVariantStyles} from './styles';
-import {getBadgeSpanProps, isFocusableElement} from './utils';
-import {BadgeWrapper} from './BadgeWrapper';
+import {badgeVariantStyles, getBadgeAnchorStyles, getBadgeButtonStyles} from './styles';
 
 export type {BadgeProps};
 
 export const Badge = React.forwardRef<HTMLElement, BadgeProps>(
-  ({variant, children, element = 'BADGE', ...props}, ref) => {
+  ({as, href, variant, children, element = 'BADGE', ...props}, ref) => {
     const resizedChildren = useResizeChildIcons(children);
 
-    const variantStyles = badgeVariantStyles[variant];
-    const spanProps = getBadgeSpanProps(props);
-    const isFocusable = isFocusableElement(props);
+    let badgeStyles = badgeVariantStyles[variant];
+
+    if (as != null) {
+      if (as === 'a') {
+        if (href === null || href === undefined)
+          throw new Error('[Paste: Badge] Badge is being used as an anchor (`as="a"`). Provide an href.');
+        if (props.onClick != null)
+          throw new Error('[Paste: Badge] An onClick was provided. To use as a button, use `as="button"`.');
+        badgeStyles = {...badgeStyles, ...getBadgeAnchorStyles()};
+      }
+      if (as === 'button') {
+        if (props.onClick === null || props.onClick === undefined)
+          throw new Error('[Paste: Badge] Badge is being used as an button (`as="button"`). Provide an onClick.');
+        if (href === null) throw new Error('[Paste: Badge] An href was provided. To use as an anchor, use `as="a"`.');
+        badgeStyles = {...badgeStyles, ...getBadgeButtonStyles(variant)};
+      }
+    }
```

Comment:
suggestion: this is getting a little hard to read. As a little inspiration you might consider how we handle this in Button, where the prop validation is pulled out into it's own little function. https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/button/src/index.tsx#L65

That might help keep the badge styling a more obvious and seperate from validation.